### ISSUE TITLE
Mod Incompatibility with Ad Astra's Planets with Shaders

### ISSUE THREAD

**Issue by ConqAra** [IrisShaders/Iris#1724]
### What happened?

[Reported on Ad Astra's Github as well](https://github.com/terrarium-earth/Ad-Astra/issues/35)

When traveling to, and landing on, planets (from Ad Astra) you will crash if you have shaders on (tested with BSL Shaders).

### Log output

[latest.log](https://github.com/IrisShaders/Iris/files/9905712/latest.log)

### Minecraft Version

1.18.2

### Iris Version

iris-mc1.18.2-1.4.0

### Operating System

Windows 10

### What is your GPU?

Nvidia GeForce RTX 3080

**Comment by coderbot16** [comment]
Relevant stack trace:

```
java.lang.NullPointerException: Cannot invoke "net.irisshaders.iris.api.v0.item.IrisItemLightProvider.getLightEmission(net.minecraft.entity.player.PlayerEntity, net.minecraft.item.ItemStack)" because the return value of "net.minecraft.item.ItemStack.getItem()" is null
	at net.coderbot.iris.uniforms.IdMapUniforms$HeldItemSupplier.update(IdMapUniforms.java:82) ~[iris-mc1.18.2-1.4.0.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at net.coderbot.iris.uniforms.FrameUpdateNotifier.onNewFrame(FrameUpdateNotifier.java:18) ~[iris-mc1.18.2-1.4.0.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.beginLevelRendering(NewWorldRenderingPipeline.java:671) ~[iris-mc1.18.2-1.4.0.jar:?]
	at net.minecraft.client.render.WorldRenderer.handler$gcf000$iris$beginLevelRender(WorldRenderer.java:13949) ~[client-intermediary.jar:?]
	at net.minecraft.client.render.WorldRenderer.render(WorldRenderer.java:1222) ~[client-intermediary.jar:?]
	at net.minecraft.client.render.GameRenderer.renderWorld(GameRenderer.java:1034) ~[client-intermediary.jar:?]
	at net.minecraft.client.render.GameRenderer.render(GameRenderer.java:813) ~[client-intermediary.jar:?]
	at net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1122) ~[client-intermediary.jar:?]
	at net.minecraft.client.MinecraftClient.run(MinecraftClient.java:737) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:236) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.10.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.10.jar:?]
	at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:577) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
```

ItemStack.getItem() is returning `null`. This looks like a potential bug in https://github.com/IrisShaders/Iris/pull/1667 by @parzivail.

**Comment by parzivail** [comment]
I thought, outside of a mixin into `ItemStack`, it was impossible for vanilla to return `null` from `getItem`? Either way, the code from #1667 was merged but is currently disabled, so it's likely code from #1563 -- it's been modified after it was merged but it should be a simple fix patch for whoever has worked on that system most recently

**Comment by ConqAra** [comment]
Fixed in Iris 1.4.1

**Comment by IMS212** [comment]
Not actually fixed in 1.4.1, delayed to 1.4.3.
